File: bit_manipulation/0231_power_of_two/0231_power_of_two.py
# -*- coding: utf-8 -*-

class Solution:
    def isPowerOfTwo(self, n):
        if n < 1:
            return False
        
        while n > 1:
            if n % 2 == 1:
                return False
            n >>= 1
        return True

print(Solution().isPowerOfTwo(5))

File: bit_manipulation/0231_power_of_two/0231_power_of_two.go
package main

import "fmt"

func isPowerOfTwo(n int) bool {
	if n < 1 {
		return false
	}

	for n > 1 {
		if n%2 == 1 {
			return false
		}
		n >>= 1
	}

	return true
}

func main() {
	n := 5
	fmt.Println(isPowerOfTwo(n))
	n = 8
	fmt.Println(isPowerOfTwo(n))
}
